code/modules/chemistry/Reagents-FoodDrink.dm

```diff
@@ -2552,12 +2552,27 @@ datum
 			thirst_value = 0.5
 			hunger_value = 1
 			viscosity = 0.5
+			var/ticks = 0
+
 			on_mob_life(var/mob/M, var/mult = 1)
 				if(M.bodytemperature < M.base_body_temp) // So it doesn't act like supermint
 					M.bodytemperature = min(M.base_body_temp, M.bodytemperature+(5 * mult))
 				M.reagents.add_reagent("sugar", 0.8 * mult)
 				if (ispug(M))
 					M.changeStatus("poisoned", 8 SECONDS * mult)
+				else if (istype(M, /mob/living/critter/small_animal/dog))
+					ticks += mult
+					switch (ticks)
+						if (1 to 15)
+							if (prob(5))
+								M.visible_message("<span class='notice'>[M] looks horribly sick! What did they eat?!</span>")
+						if (11 to 40)
+							if (prob(10))
+								M.visible_message("<span class='notice'>[M] suddenly vomits everywhere! They are clearly unwell.</span>")
+								M.vomit()
+						if (40 to INFINITY)
```

Comment:
Weird that this instantly kills dogs at a certain tick threshold instead of just doing damage. Could even just apply the `poisoned` status which pugs use (right above this block, even).

Reply:
done, just replaced it with poison. It plays well enough.